kong/plugins/datadog/handler.lua

```diff
@@ -2,143 +2,158 @@ local BasePlugin       = require "kong.plugins.base_plugin"
 local basic_serializer = require "kong.plugins.log-serializers.basic"
 local statsd_logger    = require "kong.plugins.datadog.statsd_logger"
 
-
 local ngx_log       = ngx.log
 local ngx_timer_at  = ngx.timer.at
 local string_gsub   = string.gsub
 local pairs         = pairs
 local string_format = string.format
 local NGX_ERR       = ngx.ERR
 
-
 local DatadogHandler    = BasePlugin:extend()
 DatadogHandler.PRIORITY = 10
 DatadogHandler.VERSION = "0.1.0"
 
-
-local get_consumer_id = {
-  consumer_id = function(consumer)
-    return consumer and string_gsub(consumer.id, "-", "_")
-  end,
-  custom_id   = function(consumer)
-    return consumer and consumer.custom_id
-  end,
-  username    = function(consumer)
-    return consumer and consumer.username
-  end
+local consumer_id_functions = {
+  consumer_id = function(consumer) return consumer and string_gsub(consumer.id, "-", "_") end,
+  custom_id   = function(consumer) return consumer and consumer.custom_id end,
+  username    = function(consumer) return consumer and consumer.username end
 }
 
+local function get_consumer_id(metric_config, message)
+  local consumer_id_func = consumer_id_functions[metric_config.consumer_identifier]
+  return consumer_id_func(message.consumer)
+end
+
+local metric_functions = {}
```

Comment:
Again unnecessary change, we know all the functions of this table and constructor should include all those functions. Like it was done before. I also don't see any reason to change the name.